assigner/tests/config_test.py

```diff
@@ -0,0 +1,92 @@
+from assigner.tests.utils import AssignerTestCase
+
+from assigner.config.versions import validate, get_version, upgrade, ValidationError
+from assigner.config.upgrades import UPGRADES
+from assigner.config.schemas import SCHEMAS
+
+
+CONFIGS = [
+    { # Version 0
+        "token": "xxx gitlab token xxx",
+        "gitlab-host": "https://git.gitlab.com",
+        "namespace": "assigner-testing",
+        "semester": "2016-SP",
+        "roster": [],
+    },
+    { # Version 1
+        "gitlab-token": "xxx gitlab token xxx",
+        "gitlab-host": "https://git.gitlab.com",
+        "namespace": "assigner-testing",
+        "semester": "2016-SP",
+        "roster": [],
+    },
+    { # Version 2
+        "version": 2,
+        "backend": {
+            "name": "gitlab",
+            "token": "xxx gitlab token xxx",
+            "host": "https://git.gitlab.com",
+        },
+        "namespace": "assigner-testing",
+        "semester": "2016-SP",
+        "roster": [],
+    },
+]
+
+
+EMPTY_CONFIGS = [
+    {},
+    {},
+    {
+        "version": 2,
+        "backend": {
+            "name": "gitlab",
+        },
+    },
+]
+
+
+class UpgradeTester(AssignerTestCase):
+    def test_that_we_are_testing_all_schemas_and_upgrades(self):
+        self.assertEqual(len(CONFIGS), len(SCHEMAS))
+        self.assertEqual(len(CONFIGS), len(UPGRADES) + 1)
+        self.assertEqual(len(CONFIGS), len(EMPTY_CONFIGS))
+
+    def test_get_version(self):
+        for version, config in enumerate(CONFIGS):
+            self.assertEqual(version, get_version(config))
+
+    def test_validate(self):
+        for version, config in enumerate(CONFIGS):
+            try:
+                validate(config, version)
```

Comment:
Two alternatives:
```python
with self.assertRaises(ValidationError):
    validate(config, version)
```
or

```python
self.assertRaises(validate, config, version)
```

I prefer the former.

Reply:
Wait, I want the opposite of `assertRaises`. Does such a thing exist?

Reply:
Oh, yeah, just run the function :DD. If an exception is thrown, BAM. Assuming you're doing try: ... except: ... self.fail, that's fine.